### PR TITLE
coreboot-configurator: init at 2022-08-22

### PR DESCRIPTION
**Pull request by danth**
###### Description of changes

Packaged [Coreboot Configurator](https://support.starlabs.systems/kb/guides/coreboot-configurator).

> A simple GUI to change settings in coreboot's CBFS, via the nvramtool utility.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -3406,6 +3406,8 @@ with pkgs;
     acpidump-all
     coreboot-utils;
 
+  coreboot-configurator = callPackage ../tools/misc/coreboot-configurator { };
```
Comment (azahi) [inline pkgs/top-level/all-packages.nix:3409]:
```suggestion
  coreboot-configurator = libsForQt5.callPackage ../tools/misc/coreboot-configurator { };
```

File: pkgs/tools/misc/coreboot-configurator/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchFromGitHub, inkscape, meson, ninja, pkg-config, libyamlcpp, nvramtool, qt5 }:
```
Comment (azahi) [inline pkgs/tools/misc/coreboot-configurator/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, inkscape
, meson
, ninja
, pkg-config
, libyamlcpp
, nvramtool
, wrapQtAppsHook
, qtbase
, qtsvg
}:
```
Also, is Inkscape really required here?

File: pkgs/tools/misc/coreboot-configurator/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  version = "2022-08-22";
+
+  src = fetchFromGitHub {
+    owner = "StarLabsLtd";
+    repo = "coreboot-configurator";
+    rev = "37c93e7e101a20f85be309904177b9404875cfd8";
+    sha256 = "2pk+uJk1EnVNO2vO1zF9Q6TLpij69iRdr5DFiNcZlM0=";
+  };
+
+  nativeBuildInputs = [ inkscape meson ninja pkg-config qt5.wrapQtAppsHook ];
```
Comment (azahi) [inline pkgs/tools/misc/coreboot-configurator/default.nix:14]:
```suggestion
  nativeBuildInputs = [ inkscape meson ninja pkg-config wrapQtAppsHook ];
```

File: pkgs/tools/misc/coreboot-configurator/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  buildInputs = [ libyamlcpp qt5.qtbase qt5.qtsvg ];
```
Comment (azahi) [inline pkgs/tools/misc/coreboot-configurator/default.nix:15]:
```suggestion
  buildInputs = [ libyamlcpp qtbase qtsvg ];
```

File: pkgs/tools/misc/coreboot-configurator/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
+{ lib, stdenv, fetchFromGitHub, inkscape, meson, ninja, pkg-config, libyamlcpp, nvramtool, qt5 }:
+
+stdenv.mkDerivation {
+  pname = "coreboot-configurator";
+  version = "2022-08-22";
```
Comment (azahi) [inline pkgs/tools/misc/coreboot-configurator/default.nix:5]:
```suggestion
  version = "unstable-2022-08-22";
```

File: pkgs/tools/misc/coreboot-configurator/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    rev = "37c93e7e101a20f85be309904177b9404875cfd8";
+    sha256 = "2pk+uJk1EnVNO2vO1zF9Q6TLpij69iRdr5DFiNcZlM0=";
+  };
+
+  nativeBuildInputs = [ inkscape meson ninja pkg-config qt5.wrapQtAppsHook ];
+  buildInputs = [ libyamlcpp qt5.qtbase qt5.qtsvg ];
+
+  postPatch = ''
+    substituteInPlace src/application/*.cpp \
+      --replace '/usr/bin/pkexec' '/run/wrappers/bin/sudo' \
```
Comment (azahi) [inline pkgs/tools/misc/coreboot-configurator/default.nix:19]: This largely assumes that the user uses `sudo`, and not `doas` or similar. Is there another way we can do this without hardcoding this stuff?

File: pkgs/tools/misc/coreboot-configurator/default.nix [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      --replace '/usr/bin/systemctl' '/run/current-system/sw/bin/systemctl' \
+      --replace '/usr/sbin/nvramtool' '${nvramtool}/bin/nvramtool'
+  '';
+
+  meta = with lib; {
```
Comment (azahi) [inline pkgs/tools/misc/coreboot-configurator/default.nix:24]: Missing `meta.platforms`.

File: pkgs/tools/misc/coreboot-configurator/default.nix [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    description = "A simple GUI to change settings in Coreboot's CBFS";
+    homepage = "https://support.starlabs.systems/kb/guides/coreboot-configurator";
+    license = licenses.gpl2;
```
Comment (azahi) [inline pkgs/tools/misc/coreboot-configurator/default.nix:27]: https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/coreboot-configurator/default.nix (hunk 6 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/coreboot-configurator/default.nix:19]: Also this only works on NixOS and not when using nixpkgs on other distros.

File: pkgs/tools/misc/coreboot-configurator/default.nix (hunk 2 above)
Comment (danth) [reply pkgs/tools/misc/coreboot-configurator/default.nix]: Yes, it's used to generate some icons:

https://github.com/StarLabsLtd/coreboot-configurator/blob/37c93e7e101a20f85be309904177b9404875cfd8/src/resources/meson.build#L22-L43

File: pkgs/tools/misc/coreboot-configurator/default.nix (hunk 6 above)
Comment (danth) [reply pkgs/tools/misc/coreboot-configurator/default.nix:31]: The application uses a hardcoded path so we need to choose one to substitute in.

I'm not sure how we can make it work simultaneously on NixOS and other distros, as they will require a different path.

Comment (SuperSandro2000) [reply pkgs/tools/misc/coreboot-configurator/default.nix:31]: Do we really must use a full path and cannot rely to resolve this via PATH?

Comment (danth) [reply pkgs/tools/misc/coreboot-configurator/default.nix]: Actually it does work using PATH. Good suggestion.

**Pull request merged**